types/types.go

```diff
@@ -0,0 +1,8 @@
+package types
+
+// Pr is an interface which is implemented by all proxies
+type Pr interface {
```

Comment:
conflicting feelings about the name of this interface. Runner? maybe even... Server?

Reply:
Runner it is